diff --git a/src/mqtt_codec.c b/src/mqtt_codec.c
index e8a1ca7..eac004f 100644
--- a/src/mqtt_codec.c
+++ b/src/mqtt_codec.c
@@ -485,6 +485,7 @@ static int constructConnPayload(BUFFER_HANDLE ctrlPacket, const MQTT_CLIENT_OPTI
                 if (trace_log != NULL)
                 {
                     (void)STRING_sprintf(connect_payload_trace, " | PWD: XXXX");
+                    //(void)STRING_sprintf(connect_payload_trace, " | PWD: %s", mqttOptions->password); //print PWD for debug purpose
                 }
             }
             // TODO: Get the rest of the flags
